### ISSUE TITLE
video modes are becoming a mess

### ISSUE THREAD

**Issue by mdhoney** [AppleWin/AppleWin#852]
As of the latest version, you can choose the following video modes (Video.cpp):

		TEXT("Monochrome (Custom)\0")
		TEXT("Color (Composite Idealized)\0")		// newly added
		TEXT("Color (RGB Card/Monitor)\0")	// was "Color (RGB Monitor)"
		TEXT("Color (Composite Monitor)\0")	// was "Color (NTSC Monitor)"
		TEXT("Color TV\0")
		TEXT("B&W TV\0")
		TEXT("Monochrome (Amber)\0")
		TEXT("Monochrome (Green)\0")
		TEXT("Monochrome (White)\0")

I think the menu is becoming overloaded, and somewhat confusing.  RGB and Composite Idealized look almost identical in v.1.29.15.0.  Also, can we move the "Monochrome (Custom)" to the bottom of this list?

**Comment by tomcw** [comment]
>video modes are becoming a mess
>I think the menu is becoming overloaded, and somewhat confusing.

Other than move "Monochrome (Custom)" to the bottom of this list (which I agree with) what else do you suggest doing? Moving 1 item from the top to the bottom doesn't really solve anything.

**Comment by sicklittlemonkey** [comment]
Ideally this list would customisable. For instance, being able to
enable/disable each entry. But yeah, that means another tab ...

**Comment by Lisias** [comment]
> RGB and Composite Idealized look almost identical in v.1.29.15.0. 

Nope. RGB has some additional (color) text modes (somewhat similar to the IIgs). They are not the same.

**Comment by mdhoney** [comment]
> Nope. RGB has some additional (color) text modes (somewhat similar to the IIgs). They are not the same.

Can you give a good example?  I was using the intro to the Bard's Tale which might not be the best test...

**Comment by tomcw** [comment]
Bard's Tale's intro is HGR... and RGB HGR is virtually the same as the Idealized HGR (just no half-dot shift in RGB video mode).

Try looking at a DHGR game: for RGB DHGR this is rendered as 140x192 "chunky" pixels, unless one of the new RGB modes has been software-selected (eg. B&W or Mixed Colour/B&W).

Beyond this as @Lisias says there are additional colour text modes, see this issue for examples screenshots: #826.

But it seems like we aren't doing a good job of advertising what AppleWin is now capable of. What I mean is that the AppleWin.chm (help file) has no screenshots, so from a causal read you can't readily see what kind of images you'll get with the different video modes. Would some screenshots added to AppleWin.chm help?

**Comment by Michaelangel007** [comment]
@tomcw Feel free to add these images to the help.

1. Composite Idealized
 * ![1_Composite_Idealized](https://user-images.githubusercontent.com/7876796/97717313-e4672380-1a81-11eb-94e1-f4bec17e0f01.png)

2. RGB Card
 * ![2_RGB_Card_Moniotor](https://user-images.githubusercontent.com/7876796/97717343-eaf59b00-1a81-11eb-8b07-515493e19b46.png)

3, NTSC Monitor
 * ![3_NTSC_Monitor](https://user-images.githubusercontent.com/7876796/97717378-f6e15d00-1a81-11eb-88fe-7faa6d818a2e.png)

4. Color TV
 * ![4_Color_TV](https://user-images.githubusercontent.com/7876796/97717392-fa74e400-1a81-11eb-9eb2-3a58ce045f64.png)

5, Black and White TV
 * ![5_Black_and_White_TV](https://user-images.githubusercontent.com/7876796/97717412-ff399800-1a81-11eb-9f41-31d11e8e425c.png)

6, Monochrome Amber
 * ![6_Monochrome_Amber](https://user-images.githubusercontent.com/7876796/97717435-03fe4c00-1a82-11eb-9c6f-dbbd6f5629c4.png)

7. Monochrome Green
 * ![7_Monochrome_Green](https://user-images.githubusercontent.com/7876796/97717450-095b9680-1a82-11eb-8ce2-3d67fadb4f8a.png)

8. Monochrome White
 * ![8_Monochrome_White](https://user-images.githubusercontent.com/7876796/97717458-0d87b400-1a82-11eb-82e4-a27c2ae81ff8.png)

9. Monochrome Custom
 * ![9_Monochrome_Custom](https://user-images.githubusercontent.com/7876796/97717472-111b3b00-1a82-11eb-9929-1ca385acc27f.png)


Do we need/want to show a regular HGR image as well?
i.e.

* Aztec
* Archon

**Comment by Michaelangel007** [comment]
@sicklittlemonkey Agreed that this should be customizable and regrettably another dialog tab.

UI mock-up:
```
Video Monitor/Modes:

|Composite Idealized|  [-->]  |NTSC Monitor    |
|RGB Card           |         |Monochrome Green|
|Color TV           |  [<--]  |                |
|Black and White TV |         |                |
|Monochrome Amber   |         |                |
|Monochrome White   |         |                |
|Monochrome Custom  |         |                |
```

**Comment by Michaelangel007** [comment]
@tomcw Aztec HGR

1. Composite Idealized
 * ![1_Composite_Idealized](https://user-images.githubusercontent.com/7876796/97750217-9bc55f80-1aad-11eb-857a-d3b97aa911d2.png)


2. RGB Card
 * ![2_RGB_Card](https://user-images.githubusercontent.com/7876796/97750229-a08a1380-1aad-11eb-8c42-b998762ff6ba.png)

3. NTSC Monitor
 * ![3_NTSC_Monitor](https://user-images.githubusercontent.com/7876796/97750252-a7b12180-1aad-11eb-94f4-ffc6a4d93a4f.png)

4. Color TV
 * ![4_Color_TV](https://user-images.githubusercontent.com/7876796/97750269-b3044d00-1aad-11eb-9c2f-4007011d399b.png)

5. Black and White TV
 * ![5_Black_and_White_TV](https://user-images.githubusercontent.com/7876796/97750277-b4ce1080-1aad-11eb-831e-d8d0aaa4de39.png)

6. Monochrome Amber
 * ![6_Monochrome_Amber](https://user-images.githubusercontent.com/7876796/97750324-ca433a80-1aad-11eb-8029-92c17c7a343b.png)

7. Monochrome Green
 * ![7_Monochrome_Green](https://user-images.githubusercontent.com/7876796/97750326-cc0cfe00-1aad-11eb-9ab1-8cc2cfbd434d.png)

8. Monochrome White
 * ![8_Monochrome_White](https://user-images.githubusercontent.com/7876796/97750333-cfa08500-1aad-11eb-87b4-d174664e9698.png)

9. Monochrome Custom
 * ![9_Monochrome_Custom](https://user-images.githubusercontent.com/7876796/97750338-d29b7580-1aad-11eb-871c-54cbcd2792ae.png)

**Comment by sicklittlemonkey** [comment]
A very good image to compare the modes is Airheart's title screen.

**Comment by tomcw** [comment]
Thanks @Michaelangel007 (I'll redo the Aztec / "Composite Idealized" so that 'Vertical Blend' is enabled).

@sicklittlemonkey - yeah maybe Airheart's title (or Prince of Persia's title) is a more well recognised DHGR image. But I think I'll only use one set of images, so as not to overwhelm anyone reading the help.

**Comment by mdhoney** [comment]
Thanks @Michaelangel007 and @tomcw - here are the same modes with Airheart's (DHGR) title screen.  Feel free to use.

- Composite Idealized:
![AppleWin - Airheart (Composite Idealized)](https://user-images.githubusercontent.com/8086020/97806606-9d536c80-1c00-11eb-8273-190fb84a6c95.png)

- RGB Card:
![AppleWin - Airheart (RGB)](https://user-images.githubusercontent.com/8086020/97806652-e3a8cb80-1c00-11eb-89bb-7446ef58380e.png)

- Composite Monitor:
![AppleWin - Airheart (Composite Monitor)](https://user-images.githubusercontent.com/8086020/97806673-fd4a1300-1c00-11eb-9e3d-a0e1d4288a0e.png)

- Color TV:
![AppleWin - Airheart (Color TV)](https://user-images.githubusercontent.com/8086020/97806694-0cc95c00-1c01-11eb-8ad3-b8d52b72bb53.png)

- B&W TV:
![AppleWin - Airheart (B W TV)](https://user-images.githubusercontent.com/8086020/97806708-1ce13b80-1c01-11eb-8b27-b4a40317ecf6.png)

- Monochrome (Amber):
![AppleWin - Airheart (Monochrome, Amber)](https://user-images.githubusercontent.com/8086020/97806747-3f735480-1c01-11eb-88e3-84937f1b87cb.png)

- Monochrome (Green):
![AppleWin - Airheart (Monochrome, Green)](https://user-images.githubusercontent.com/8086020/97806755-48642600-1c01-11eb-8ca7-34cf3a702583.png)

- Monochrome (White):
![AppleWin - Airheart (Monochrome, White)](https://user-images.githubusercontent.com/8086020/97806758-4c904380-1c01-11eb-8007-1ddc1007d96d.png)

- Monochrome (Custom):
![AppleWin - Airheart (Monochrome, Custom)](https://user-images.githubusercontent.com/8086020/97806763-51ed8e00-1c01-11eb-953c-60a323f31801.png)

**Comment by mdhoney** [comment]
One more thing regarding the Bard's Tale intro.  Yes I know this is only HGR, but why do both "Composite Idealized" and "RGB Card" blend the lettering together (look at the letter m in "men") whereas "Composite Monitor" does not?

- Composite Idealized:
![AppleWin - Bard's Tale (Composite Idealized)](https://user-images.githubusercontent.com/8086020/97807200-9843ec80-1c03-11eb-909d-1dec10396506.png)

- RGB Card:
![AppleWin - Bard's Tale (RGB Card)](https://user-images.githubusercontent.com/8086020/97807206-a0039100-1c03-11eb-9651-095f75237113.png)

- Composite Monitor:
![AppleWin - Bard's Tale (Composite Monitor)](https://user-images.githubusercontent.com/8086020/97807209-a7c33580-1c03-11eb-9dc9-a80a105fd017.png)

**Comment by mdhoney** [comment]
> Other than move "Monochrome (Custom)" to the bottom of this list (which I agree with) what else do you suggest doing? Moving 1 item from the top to the bottom doesn't really solve anything.

You have a point @tomcw.   Do we actually need the "Monochrome (Custom)" setting?  It's neat but not related to any real hardware (as far as I know).  Similarly, since "Composite Idealized" was never achieved in real hardware, how about we add a checkbox "Idealized Composite Monitor" (which controls the "Composite Monitor" setting) and then remove that from the drop-down menu?  
Sorry if I didn't explain this well, happy to provide a mock-up...

**Comment by tomcw** [comment]
>One more thing regarding the Bard's Tale intro. Yes I know this is only HGR, but why do both "Composite Idealized" and "RGB Card" blend the lettering together (look at the letter m in "men") whereas "Composite Monitor" does not?

In HGR, "Composite Idealized" and "RGB Card" are virtually the same (expect there's no half-dot shift for "RGB Card"). See the AZTEC images above for the letters 'A' and 'Y' in the "PRESS ANY KEY" text.

btw, in "Composite Idealized you can also enable 'Vertical Blend' to produce new blended colours for both hats.

On to your question...

"RGB Card" will have a 2-dot (hires pixel) rolling window used to generate the colour (black, white + 2 colours), and also use the byte's high-bit to select the palette for the 2 colours (purple/green or orange/blue). The letter 'm' is formed of the bit pattern 1101011. Black is defined at '00' (2 adjacent zeros), which doesn't appear in 'm' / 1101011.

"Composite Monitor" emulated a higher bandwidth than TVs, because of this, it's able to render these black pixels (eg. hires 1101011 -> dhires pixels 11110011001111, so now we have '00' patterns present, so will get black dots). This is a fairly gross simplification, but hopefully it's good enough!

**Comment by sicklittlemonkey** [comment]
@mdhoney Thanks for the Airheart and other images.

@tomcw If the idea is too put these in the help file, I think side-by-side comparisons would be good.

But the differentiating features of each mode are definitely shown better in different images.

Composite idealized (which I don't care for, but ...) mode's vertical blend is better shown in Aztec.

But I think RGB vs composite TV horizontal detail and colour blending are better shown in Airheart.

Then again, Airheart looks surprisingly similar in composite idealized and composite TV, which really goes to show that the composite idealized served its purpose well before the NTSC emulation! : - )

**Comment by mdhoney** [comment]
> btw, in "Composite Idealized" you can also enable 'Vertical Blend' to produce new blended colours for both hats.

Yes, thanks for the reminder.  Here's the Bard's Tale intro again...

- Composite Idealized, Vertical Blend off:
![AppleWin - Bard's Tale (Composite Idealized) cr](https://user-images.githubusercontent.com/8086020/97850895-b7d92480-1c98-11eb-8861-d09e93280ef9.png)

- Composite Idealized, Vertical Blend on:
![AppleWin - Bard's Tale (Composite Idealized, Vertical Blend) cr](https://user-images.githubusercontent.com/8086020/97850917-c0315f80-1c98-11eb-8ea3-dcc1feb63c64.png)

**Comment by tomcw** [comment]
> Do we actually need the "Monochrome (Custom)" setting? It's neat but not related to any real hardware (as far as I know).

@mdhoney: I don't know, as we have no telemetry in AppleWin. Personally I never use it, so dropping it wouldn't bother me, but I bet there are some that are using it (as the feature has been around for ~20 years... it predates me!).

>Composite idealized (which I don't care for, but ...)

@sicklittlemonkey: ...a confusing name, for a "fake" mode adding confusing value :-/ Historically this (as you say) is the mode that existed before the NTSC support was added. As you know we did drop this video mode for 3 years, but I reinstated it due to much 'voice of customer' demand to restore it (see #357).

I use it, and see it as a best of many worlds, ie. crisp TEXT (40/80 col) like RGB (so not like Color TV), but with non-chunky dhires pixels so more like Color TV (ie. not like RGB), and continuous horizontal colour (unlike the Color Composite Monitor). If the Airheart image had 80-column TEXT in the bottom of the screen, then you'd see the difference.

>If the idea is too put these in the help file, I think side-by-side comparisons would be good.

Yes, I totally agree.

**Comment by Michaelangel007** [comment]
I occasionally use Monochrome custom mode. It was provided because not everyone liked the 3 hard-coded monochrome colors.

There is no need to remove it when it works.

It would be better to allow users to customize what video modes they want with the dialog box -- and default the Monochrome Custom setting to OFF. That way a user can `F9` through the video modes they _actually_ use.
